api/nodes/views.py

```diff
@@ -25,7 +26,11 @@ def get_node(self):
         obj = get_object_or_404(Node, self.kwargs[self.node_lookup_url_kwarg])
         # May raise a permission denied
         self.check_object_permissions(self.request, obj)
-        return obj
+
+        if obj.is_deleted:
```

Comment:
Yeah, we were going to do that, but because other models didn't necessarily have an is_deleted attribute, we leaned against. Maybe subclass it and do the is_deleted check after would be good.